aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -346,6 +350,25 @@ public static synchronized AWSMobileClient getInstance() {
         return singleton;
     }
 
+    /**
+     * For unit testing purposes only so we can force a new instance of
+     * mobile client to be created.
+     * @param forceGetNew True if a new instance of AWSMobileClient should be created.
+     * @return A new instance of AWSMobileClient.
+     */
+    @VisibleForTesting
+    static synchronized AWSMobileClient getInstance(boolean forceGetNew) {
+        if (forceGetNew) {
+            singleton = null;
+        }
+        return new AWSMobileClient();
+    }
+
+    @VisibleForTesting
+    void setUserPool(CognitoUserPool userpool) {
+        this.userpool = userpool;
+    }
```

Comment:
would it be possible to inject a mock userpool during initialization by adding something like
```
"CognitoUserPool": {
    "Default": {
        "PoolId": "<pool-id>",
        "AppClientId": "<client-id>",
        "Region": "<region>"
    }
},
```
during config generation? That would be preferable to adding more `@VisibleForTesting` methods.

Reply:
It would be tricky because you actually have to inject the mock after the initialize method runs. It would probably require a lot more refactoring, so I'm not sure it would be worth it.